lifelines/fitters/aalen_additive_fitter.py

```diff
@@ -141,14 +141,16 @@ def _fit_static(self, dataframe, duration_col, event_col=None,
         T = pd.Series(df[duration_col].values, index=ids)
 
         df = df.set_index(id_col)
-        pass_for_numeric_dtypes_or_raise(df)
+        self._check_values(df, T, C)
 
         ix = T.argsort()
         T, C = T.iloc[ix], C.iloc[ix]
 
         del df[duration_col]
         n, d = df.shape
         columns = df.columns
+        df = df.astype(float)
```

Comment:
I'm not sure I follow exactly what has caused the performance improvements. Is it the float? Or is it something else?

Reply:
It was the changes here: https://github.com/CamDavidsonPilon/lifelines/pull/433/files#diff-81aff105858474f333c6fc40c9edae5eR650

Reply:
I just found it. So the matrix is always positive definite, that makes sense.